autoload/youcompleteme.vim

```diff
@@ -738,6 +738,20 @@ endfunction
 
 command! YcmShowDetailedDiagnostic call s:ShowDetailedDiagnostic()
 
+function! s:FixIt()
+  " We force recompilation to ensure that the FixIt diagnostic data is not out
+  " of date with respect to the code. Without this we can get unpredicable
+  " results.
+  "
+  " note we do this silently so that we don't echo 2 messages vim. This ensures
+  " that this command doesn't require the user to hit return to dismiss the
+  " Fixit made <N> changes message
+  if s:ForceCompileSilent()
+    py ycm_state.FixIt()
+  endif
+endfunction
+
+command! YcmFixIt call s:FixIt()
```

Comment:
Yes, subcommands are processed in the server in the sense that it sends back to the client a data structure in a editor agnostic way and then the client interprets the data sent by the server and do something. Look at the various `GoTo` subcommands they send back the file, line and column to go to and then the client will go there. You can do the same for the FixIt I guess or there are some challenges that I didn't considered?